### PR TITLE
Exposed add_member_to_rank 

### PR DESCRIPTION
**Pull request by MrishoLukamba**
Exposed the extrinsic call only root origin can call.
Performed testing..
Before continuing I need clarifications If i got the instructions right @shawntabrizi 
Remaining the first task also to include it as kitchen sink

### CODE REVIEW

**Comment by MrishoLukamba** [comment]
This PR is on issue #12029

File: frame/ranked-collective/src/lib.rs [hunk 1]
```diff
@@ -471,6 +471,18 @@ pub mod pallet {
...
 		}
 
+		// Only root account can add a member to
+		#[pallet::weight(100)]
+		pub fn add_member_to_rank(
+			origin: OriginFor<T>,
+			member: T::AccountId,
+			rank: Rank
+		) -> DispatchResult {
+			let _= ensure_root(origin)?;
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:481]:
```suggestion
			ensure_root(origin)?;
```

File: frame/ranked-collective/src/lib.rs [hunk 2]
```diff
@@ -471,6 +471,18 @@ pub mod pallet {
...
+			Self::do_add_member_to_rank(member,rank)?;
+			Ok(())
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:483]:
```suggestion
			Self::do_add_member_to_rank(member,rank)
```

File: frame/ranked-collective/src/lib.rs [hunk 3]
```diff
@@ -471,6 +471,18 @@ pub mod pallet {
 			Self::do_add_member(who)
 		}
 
+		// Only root account can add a member to
+		#[pallet::weight(100)]
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:475]: need proper weights and benchmark

File: frame/ranked-collective/src/tests.rs [hunk 4]
```diff
@@ -44,10 +44,16 @@ frame_support::construct_runtime!(
...
+		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T>},
 		Club: pallet_ranked_collective::{Pallet, Call, Storage, Event<T>},
 	}
 );
 
+impl pallet_sudo::Config for Test {
+	type Event = Event;
+	type Call = Call;
+}
+
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/tests.rs:56]: you dont need this, so please remove all the stuff related to sudo

**Comment by shawntabrizi** [comment]
You should always add the new extrinsic at the bottom of the extrinsics.

**Comment by MrishoLukamba** [comment]
okey am on it, thanks

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1756211

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1756210

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1756292

**Comment by shawntabrizi** [comment]
@MrishoLukamba please keep this PR minimal and only adding the new extrinsic. you can open a new PR for the other fellowship feature

**Comment by MrishoLukamba** [comment]
This is outdated

**Pull request closed without merging**